app/assets/javascripts/stories.js

```diff
@@ -25,8 +25,57 @@ document.addEventListener("DOMContentLoaded", () => {
       debounceTimer = window.setTimeout(updateMarkdown, 300);
     });
   });
+
+  const form = document.querySelector(".edit_story");
+  const backButton = document.getElementById("back");
+  const logo = document.getElementById("logo");
+  let isDirty = false;
+
+  if (form) {
+    // Mark the form as dirty when any input changes
+    form.addEventListener("input", function () {
+      isDirty = true;
+      addBeforeUnloadEventListener(isDirty);
+    });
+
+    // Reset isDirty on form submission
+    form.addEventListener("submit", function () {
+      isDirty = false;
+      addBeforeUnloadEventListener(isDirty);
+    });
+  }
+
+  // Attach a click event to the custom back button
+  [backButton, logo].forEach(element => {
+    element.addEventListener("click", function (event) {
+      if (isDirty) {
+        const confirmLeave = confirm("You have unsaved changes. Are you sure you want to go back?");
+        if (!confirmLeave) {
+          // Prevent navigation if the user chooses not to leave
+          event.preventDefault();
+        } else {
+          // Optionally, reset isDirty if leaving
+          isDirty = false;
+          addBeforeUnloadEventListener(isDirty)
+        }
```

Comment:
I find it easier to read `confirmLeave` compared to `!confirmLeave`, as it always takes me a moment to grasp the condition.

```suggestion
        if (confirmLeave) {
          // Optionally, reset isDirty if leaving
          isDirty = false;
          addBeforeUnloadEventListener(isDirty)
        } else {
          // Prevent navigation if the user chooses not to leave
          event.preventDefault();
        }
```